shell/platform/darwin/common/BUILD.gn

```diff
@@ -13,13 +13,29 @@ source_set("common") {
   cflags_objcc = flutter_cflags_objcc
 
   sources = [
-    "availability_version_check.cc",
     "buffer_conversions.h",
     "buffer_conversions.mm",
     "command_line.h",
     "command_line.mm",
   ]
 
+  deps = [
+    ":availability_version_check",
+    "//flutter/fml",
+  ]
+
+  public_configs = [ "//flutter:config" ]
+}
+
+# Provides an implementation for _availability_version_check.
+#
+# See: https://github.com/flutter/flutter/issues/133777
```

Comment:
this source code is likely to outlive github, it would be ideal if we could inline the explanation.

Reply:
Sounds good; will do.

Reply:
Done - copied in the text from the commit log (which hopefully also outlives GitHub!).